tool-sidecar/src/unstract/tool_sidecar/log_processor.py

```diff
@@ -0,0 +1,275 @@
+the tool's output log file and streaming logs to Redis while watching for
+completion signals.
+"""
+
+import json
+import logging
+import os
+import time
+from datetime import datetime, timezone
+from typing import Any, Optional
+
+from unstract.core.constants import LogFieldName
+from unstract.core.pubsub_helper import LogPublisher
+
+from .constants import Env, LogLevel, LogType
+from .dto import LogLineDTO
+
+logging.basicConfig(
+    level=getattr(logging, os.getenv(Env.LOG_LEVEL, "INFO")),
+    format="%(asctime)s - %(name)s - %(levelname)s - %(message)s",
+)
+logger = logging.getLogger(__name__)
+
+
+class LogProcessor:
+
+    TERMINATION_MARKER = "TOOL_EXECUTION_COMPLETE"
+
+    def __init__(
+        self,
+        log_path: str,
+        redis_host: str,
+        redis_port: str,
+        redis_user: str,
+        redis_password: str,
+        tool_instance_id: str,
+        execution_id: str,
+        organization_id: str,
+        file_execution_id: str,
+        messaging_channel: str,
+    ):
+        """
+        Initialize the log processor with necessary connections and paths.
+
+        Args:
+            log_path: Path to the log file to monitor
+            redis_url: URL for Redis connection
+            tool_instance_id: ID of the tool instance
+            execution_id: ID of the execution
+            organization_id: ID of the organization
+            file_execution_id: ID of the file execution
+        """
+        self.log_path = log_path
+        self.redis_host = redis_host
+        self.redis_port = redis_port
+        self.redis_user = redis_user
+        self.redis_password = redis_password
+        self.tool_instance_id = tool_instance_id
+        self.execution_id = execution_id
+        self.organization_id = organization_id
+        self.file_execution_id = file_execution_id
+        self.messaging_channel = messaging_channel
+
+    def wait_for_log_file(self, timeout: int = 300) -> bool:
+        """
+        Wait for the log file to be created by the tool container.
+
+        Args:
+            timeout: Maximum time to wait in seconds
+
+        Returns:
+            bool: True if file exists, False if timeout occurred
+        """
+        start_time = time.time()
+        while not os.path.exists(self.log_path):
+            if time.time() - start_time > timeout:
+                return False
+            time.sleep(0.1)
+        return True
+
+    def process_log_line(self, line: str) -> LogLineDTO:
+        """
+        Process a single log line, checking for completion signal.
+
+        Args:
+            line: Log line to process
+
+        Returns:
+            Optional[Dict]: Parsed JSON if line is a completion signal
+        """
+        # Stream log to Redis
+        if self.TERMINATION_MARKER in line:
+            logger.info(
+                f"Tool container terminated with status {self.TERMINATION_MARKER}"
+            )
+            return LogLineDTO(is_completion=True)
```

Comment:
```suggestion
            return LogLineDTO(id_terminated=True)
```
I suggest renaming it to `is_complete` / `is_terminated` instead